### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by kennykerr**
First stab at documentation and some code cleanup.

### CODE REVIEW

File: src/com/interface.rs [hunk 1]
```diff
@@ -1,58 +1,56 @@
...
-///
-/// VTables must be COM compliant VTables where the first three function
-/// pointers are the IUknown methods.
-///
-/// Additionally, because ComInterfaces are just pointers to vtables,
-/// it must be safe to zero initialize the interface.
+/// `*mut *mut Self::VTable`.
 pub unsafe trait ComInterface: Sized {
+    /// `VTable` must be a COM compliant vtable where the first three function
+    /// pointers are the `IUnknown` methods. And because ComInterfaces are just
```
Comment (rylev) [inline src/com/interface.rs:12]: This last sentence is about ComInterfaces as a whole not about vtables. I think we should move it back up

File: src/com/ref_count.rs [hunk 2]
```diff
@@ -1,28 +1,37 @@
-use std::sync::atomic::{self, AtomicU32, Ordering};
+use std::sync::atomic::{self, AtomicI32, Ordering};
 
+/// A thread-safe reference count for use with COM implementations.
 #[repr(transparent)]
 pub struct RefCount {
-    value: AtomicU32,
+    value: AtomicI32,
```
Comment (rylev) [inline src/com/ref_count.rs:6]: Why is this better as an i32? Isn’t it a bug if the count goes below 0?

File: src/com/interface.rs (hunk 1 above)
Comment (kennykerr) [reply src/com/interface.rs:12]: Will do.

File: src/com/ref_count.rs (hunk 2 above)
Comment (kennykerr) [reply src/com/ref_count.rs:6]: Yes, it is a bug if the count goes below zero. Using `i32` lets us detect this and panic if the object is used after being released. That's the panic below when we release and realize we're already at zero. Hard to test for though...

**Pull request merged**